pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -495,23 +564,38 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 		if identityMatches(set, replicas[i]) && storageMatches(set, replicas[i]) && retentionMatch {
 			continue
 		}
-		// Make a deep copy so we don't mutate the shared cache
-		replica := replicas[i].DeepCopy()
-		if err := ssc.podControl.UpdateStatefulPod(ctx, updateSet, replica); err != nil {
-			return &status, err
-		}
+		actions = append(actions, func() (bool, error) {
+			// Make a deep copy so we don't mutate the shared cache
+			replica := replicas[i].DeepCopy()
+			if err := ssc.podControl.UpdateStatefulPod(ctx, updateSet, replica); err != nil {
+				return true, err
+			}
+			return false, nil
+		})
 	}
 
+	if monotonic {
+		for _, fn := range actions {
+			if shouldExit, err := fn(); shouldExit || err != nil {
+				return &status, err
```

Comment:
I think the status here may not be up to date -- e.g. if you created some pod it may require updating.

The caller is reading the status even in case of error: https://github.com/kubernetes/kubernetes/blob/cf03aac61973bcfba62721cfeaf713004c7f49c0/pkg/controller/statefulset/stateful_set_control.go#L113-L113

Reply:
Done